scalalib/overrides/scala/runtime/BoxesRunTime.scala

```diff
@@ -53,13 +53,31 @@ object BoxesRunTime {
 
   def equals(x: Object, y: Object): Boolean =
     linkTimeIf(LinkingInfo.targetPureWasm) {
-      if (x eq y) true
+      if (strictEquals(x, y)) true
```

Comment:
This initial test is meant to be a fast path. But now it's slow as well, so we don't gain anything. 

Consider still using `eq`, but specifically detect and reject `NaN` as follows: 
```scala
if (x eq y) {
  x match {
    case x: Double => x == x // rejects NaN
    case _         => true
  }
}
```